src/Model_TEST.cc

```diff
@@ -285,6 +286,40 @@ TEST(DOMModel, AddModel)
   EXPECT_EQ(modelFromModel->Name(), nestedModel.Name());
 }
 
+/////////////////////////////////////////////////
+TEST(DOMModel, AddModifyFrame)
+{
+  sdf::Model model;
+  EXPECT_EQ(0u, model.FrameCount());
+
+  sdf::Frame frame;
+  frame.SetName("frame1");
+  EXPECT_TRUE(model.AddFrame(frame));
+  EXPECT_EQ(1u, model.FrameCount());
+  EXPECT_FALSE(model.AddFrame(frame));
+  EXPECT_EQ(1u, model.FrameCount());
+
+  model.ClearFrames();
+  EXPECT_EQ(0u, model.FrameCount());
+
+  EXPECT_TRUE(model.AddFrame(frame));
+  EXPECT_EQ(1u, model.FrameCount());
+
+  const sdf::Frame *frameFromModel = model.FrameByIndex(0);
+  ASSERT_NE(nullptr, frameFromModel);
+  EXPECT_EQ(frameFromModel->Name(), frame.Name());
+
+  sdf::Frame *mutableFrame = model.FrameByIndex(0);
+  mutableFrame->SetName("newName1");
+  EXPECT_EQ(mutableFrame->Name(), model.FrameByIndex(0)->Name());
+
+  sdf::Frame *mutableFrameByName = model.FrameByName("frame1");
+  EXPECT_EQ(nullptr, mutableFrameByName);
+  mutableFrameByName = model.FrameByName("newName1");
+  ASSERT_NE(nullptr, mutableFrameByName);
+  EXPECT_EQ(mutableFrameByName->Name(), model.FrameByName("newName1")->Name());
```

Comment:
This is currently comparing `sdf::Frame::Name` against `sdf::Frame::Name`, which even if there was something wrong with the implementation, the test would pass because they will return the same thing.

I would think you want to compare the name against the known name like so:
```suggestion
  EXPECT_EQ("newName1", mutableFrameByName->Name());
```

Reply:
Change per suggestion.